hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationSupervisorMetrics.java

```diff
@@ -71,12 +72,51 @@ public void getMetrics(MetricsCollector collector, boolean all) {
         .addGauge(Interns.info("numRequestedReplications",
             "Number of requested replications"),
             supervisor.getReplicationRequestCount())
+        .addGauge(Interns.info("numRequestedECReconstructions",
```

Comment:
If a new task subclass was added in the future (I think reconciliation may be one), then the earlier code in this PR that populates the HashMaps will automatically start gathering metrics for the new class. But the code here will never present them.

Could we add all these gauges by iterating each map, so that if a new task appears in the future, it will automatically get added?

You may need to add some abstract method to AbstractTask to be able to get a meaningful name for each of the tasks that each sub class must implement.

Then instead of `requestCounter.put(task.getClass(), new AtomicLong(0));` have:

```
requestCounter.put(task.getMetricName, new AtomicLong(0));
```

The metric names may be something like:

ECReconstructions
ContainerReplications

etc.

There may be a better way of doing this too - that is the first thought I came up with.

Reply:
Thanks @sodonnel  for the comment and review.
I think this is a great idea, I will update it later.